apps/shared/test-helpers/mine.js

```diff
@@ -1,43 +1,49 @@
-const Web3 = require("web3");
-const PrivateKeyProvider = require("truffle-privatekey-provider");
-const provider = new PrivateKeyProvider(
-    "A8A54B2D8197BC0B19BB8A084031BE71835580A01E70A45A13BABD16C9BC1563",
-    "http://localhost:8545"
-  );
-const web3 = new Web3(provider);
+/// uncomment the Below block to use on aragon devchain without truffle
 
-const mineBlock = require("./mineBlock")(web3)
-const getBlock = require("./blockNumber")(web3)
+//const Web3 = require("web3");
+//const PrivateKeyProvider = require("truffle-privatekey-provider");
+//const provider = new PrivateKeyProvider(
+//    "A8A54B2D8197BC0B19BB8A084031BE71835580A01E70A45A13BABD16C9BC1563",
+//    "http://localhost:8545"
+//  );
+//const web3 = new Web3(provider);
 
-const mineBlocks = async (blocks) => {
-    for (var i = 0; i < blocks; i++) {
-        try {
-            await mineBlock()
-        }
-        catch(e) {
-            console.error('error: ',e)
+const mineBlock = require("./mineBlock")
+const getBlock = require("./blockNumber")
+
+module.exports = async (cb) => {
+    const mineBlockFn = mineBlock(web3)
+    const getBlockFn = getBlock(web3)
+
+    const mineBlocks = async (blocks) => {
+        for (var i = 0; i < blocks; i++) {
+            try {
+                await mineBlockFn()
```

Comment:
We never use the `i` variable. I expected this `mineBlockFn` to receive a given block as an argument. How does it work, given that it receives no arguments? Does it do the same thing every single time?

Is there a way to use `Promise.all` so we don't have to await one `mineBlockFn` to complete in order to start the next one?

I see this isn't new code, but while we're here, can we use `blocks.forEach(block => {` instead of the `for` statement?

Reply:
I can functionalize this, but mineBlockFn is a very primitive rpc command that accepts no arguments and just commands the devchain to mine a single block. The whole purpose of this script is to be able to run commands like:
`npm run mine to x` as in mine to block x and `npm run mine x` as in mine x number of blocks

Reply:
Hmm, gotcha. So we could use `blocks.forEach(() =>` here, since we don't actually care about any of the info associated with a particular block.